File: CS50-Havard-2018/pset6/cash/cash.py
import cs50

while True:
    print("Change owed: ")
    amount = cs50.get_float()
    if amount >= 0:
        break
# num is the number of coins
num = 0
cents = int(round(amount * 100))
coins = [25, 10, 5, 1]
for coin in coins:
    num += cents // coin
    cents %= coin
print("{}".format(num))

File: Harvard_CS50/pset/pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
/* This question aims to calculuate the mimimum number of coins to change
by using greedy algorithm. Four types of coins are used: quarter, dime, nickel
and penny
*/

int main(void)
{
    float dollars;
    do
    {
        printf("Change owed: ");
        dollars = get_float();
    }
    while (dollars < 0);
    // dollars to cents and round
    unsigned cents = round(dollars * 100);
    unsigned coin_count = 0;
    unsigned coins[4] = {25, 10, 5, 1};
    // loop over 4 types of coins and add to toal coin count
    for (int i = 0; i < 4; ++i)
    {
        coin_count += cents / coins[i];
        cents %= coins[i];
    }
    // print the result
    printf("%i\n", coin_count);
    return 0;
}
